apps/aviatorclk/aviatorclk.app.js

```diff
@@ -0,0 +1,283 @@
+function drawSeconds() {
+  let now = new Date();
+  let seconds = now.getSeconds().toString();
+  if (seconds.length == 1) seconds = '0' + seconds;
+  let y = Bangle.appRect.y + mainTimeHeight - 3;
+  g.setFontAlign(-1, 1).setFont("Vector", secondaryFontHeight).setColor(COLOUR_GREY);
+  g.drawString(seconds, horizontalCenter + 54, y, true);
+}
+
+// sync seconds update
+function syncSecondsUpdate() {
+  drawSeconds();
+  setTimeout(function() {
+    drawSeconds();
+    secondsInterval = setInterval(drawSeconds, 1000);
+  }, 1000 - (Date.now() % 1000));
+}
+
+// set timeout for per-minute updates
+function queueDraw() {
+  if (drawTimeout) clearTimeout(drawTimeout);
+  drawTimeout = setTimeout(function() {
+    drawTimeout = undefined;
+    if (METARts) {
+      let now = new Date();
+      let METARage = Math.floor((now - METARts) / 60000);
+      if (METARage > 60) {
+        // the METAR colour might have to be updated:
+        drawAVWX();
+      }
+    }
+    draw();
+  }, 60000 - (Date.now() % 60000));
+}
+
+// draw top part of clock (main time, date and UTC)
+function draw() {
+  let now = new Date();
+  let nowUTC = new Date(now + (now.getTimezoneOffset() * 1000 * 60));
+
+  // prepare main clock area
+  let y = Bangle.appRect.y;
+
+  g.setBgColor(g.theme.bg);
+
+  // main time display
+  g.setFontAlign(0, -1).setFont("Vector", mainTimeHeight).setColor(g.theme.fg);
+  g.drawString(timeStr(now, false), horizontalCenter, y, true);
+
+  // prepare second line (UTC and date)
+  y += mainTimeHeight;
+  g.clearRect(0, y, g.getWidth(), y + secondaryFontHeight - 1);
+
+  // weekday and day of the month
+  g.setFontAlign(-1, -1).setFont("Vector", secondaryFontHeight).setColor(dateColour);
+  g.drawString(require("locale").dow(now, 1).toUpperCase() + ' ' + now.getDate(), 0, y, false);
+
+  // UTC
+  g.setFontAlign(1, -1).setFont("Vector", secondaryFontHeight).setColor(UTCColour);
+  g.drawString(timeStr(nowUTC, false) + "Z", g.getWidth(), y, false);
+
+  queueDraw();
+}
+
+
+// initialise
+g.clear(true);
+
+// scroll METAR lines on taps
+Bangle.setUI("clockupdown", action => {
+  switch (action) {
+    case -1:  // top tap
+      if (settings.invertScrolling) {
+        if (METARscollLines > 0)
+          METARscollLines--;
+      } else {
+        if (METARscollLines < METARlinesCount - 4)
+          METARscollLines++;
+      }
+      break;
+    case 1:   // bottom tap
+      if (settings.invertScrolling) {
+        if (METARscollLines < METARlinesCount - 4)
+          METARscollLines++;
+      } else {
+        if (METARscollLines > 0)
+          METARscollLines--;
+      }
+      break;
+    default:
+      // ignore
+  }
+  drawAVWX();
+});
+
+// load widgets
+Bangle.loadWidgets();
+Bangle.drawWidgets();
+
+// draw static separator line
+y = Bangle.appRect.y + mainTimeHeight + secondaryFontHeight;
+g.setColor(separatorColour);
+g.drawLine(0, y, g.getWidth(), y);
+
+// draw times and request METAR
+draw();
+if (settings.showSeconds)
```

Comment:
You might only want to draw settings if unlocked, else display updates can drain the battery faster

Reply:
I assume you mean "draw seconds" (instead of "settings" - please correct me if I'm wrong).

If the seconds display is required or useful in flight (and therefore enabled through the settings), it's actually handy to have it constantly available (ie. without unlocking). I've also tried to minimise the code for the seconds updates (only the 2 seconds digits are updated every second).

I am considering implementing a way to "dynamically" enable the seconds display (double tap, or create 3 "tap zones" for scroll up / down and enable seconds display, etc.).

Reply:
right, its "seconds". Just wanted to make sure that you are aware of the consequences to battery life.

Reply:
Thanks - yes, I'm aware of the extra power draw.
The readme points this out to the end user:
```
- **Show Seconds**: to conserve battery power, you can turn the seconds display off
```